CodeGeneration/Sources/generate-swiftsyntaxbuilder/ResultBuildersFile.swift

```diff
@@ -0,0 +1,194 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SwiftSyntax
+import SyntaxSupport
+import SwiftSyntaxBuilder
+import Utils
+
+let resultBuildersFile = SourceFile {
+  ImportDecl(
+    leadingTrivia: .docLineComment(copyrightHeader),
+    path: "SwiftSyntax"
+  )
+  for node in SYNTAX_NODES where node.isSyntaxCollection {
+    let type = SyntaxBuildableType(syntaxKind: node.syntaxKind)
+    let elementType = node.collectionElementType
+    
+    StructDecl(
+      attributes: Token.identifier("@resultBuilder").withLeadingTrivia(.newlines(1)).withTrailingTrivia(.newlines(1)),
+      modifiers: Token.public,
+      identifier: "\(type.syntaxKind)Builder") {
+        
+        TypealiasDecl(
+          """
+          /// The type of individual statement expressions in the transformed function,
+          /// which defaults to Component if buildExpression() is not provided.
+          public typealias Expression = \(elementType.expressibleAsBaseName)
+          """
+        )
+        
+        TypealiasDecl(
+          """
+          /// The type of a partial result, which will be carried through all of the
+          /// build methods.
+          public typealias Component = [\(elementType.expressibleAs)]
+          """
+        )
+        
+        TypealiasDecl(
+          """
+          /// The type of the final returned result, which defaults to Component if
+          /// buildFinalResult() is not provided.
+          public typealias FinalResult = \(type.buildable)
+          """
+        )
+        
+        FunctionDecl(
+          """
+          /// Required by every result builder to build combined results from
+          /// statement blocks.
+          public static func buildBlock(_ components: Component...) -> Component {
+            return components.flatMap { $0 }
```

Comment:
Since all of these functions return single expressions, we could omit the `return` here (and in the other builder functions).

Reply:
I have a general preference for explicit `return`, but that might be personal. I gave up actively fighting against implicit returns 😆

Reply:
Personally I also like returns in methods as it read better IMO 😬